### PR TITLE
Proper component chaining and display using web-socket

### PR DESCRIPTION
**Pull request by john-langlois**
-Changed from having multiple components obtaining different data for gps, imu, enviro, etc. To having one component that would obtain the necessary information to display in the status component of the ground station

-Moved the function that obtains the data from the server through the WebSocket to the App component (top of the chain), to be able to be accessed from any other component through function parameters.

-Added conditional statements for error catching when the data is undefined, which will show 0 instead.

### CODE REVIEW

File: dockerfiles/Dockerfile.react [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:16
+FROM node:17
```
Comment (jeffstjean) [inline dockerfiles/Dockerfile.react:1]: Change to `FROM node:16`

Noticed some things breaking after upgrading

File: frontend/package.json [hunk 2]
```diff
@@ -12,8 +12,10 @@
     "@testing-library/user-event": "^13.5.0",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
-    "react-scripts": "5.0.0",
-    "web-vitals": "^2.1.3"
+    "react-scripts": "^2.1.3",
```
Comment (jeffstjean) [inline frontend/package.json]: We were working on version 5.0.0. Not sure how it got downgraded to 2.1.3. Change this to `"react-scripts": "5.0.0",` and we'll see if it causes any issues

File: frontend/src/components/Telemetry.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  return(
+
+      <TelemetryStatus
+      telemetry = {telemetryData}
+      />
+    
+  )
+}
+export default Telemetry;
```
Comment (jeffstjean) [inline frontend/src/components/Telemetry.js]: Do we need this entire component if it just returns the TelemetryStatus component? Might be missing something but seems like this is redundant?

File: package-lock.json [hunk 4]
```diff
@@ -1,6 +1,2322 @@
 {
```
Comment (jeffstjean) [inline package-lock.json:1]: This entire file should be deleted along with the `package.json` in the root directory (`package-lock.json` and `package.json` should only be in the `api/`, `frontend/`, and `usb/` directories

**Review by jeffstjean — CHANGES_REQUESTED** [review]
Work through these four changes and rebase from develop onto your branch to get the most recent changes before we merge this into develop. See the [new git into tutorial](https://github.com/UWO-Aero-Design/first-commit/tree/master/git-intro) or message me if you need help rebasing.

**Comment by jeffstjean** [comment]
CSS isn't a priority here but a lot of the values are really hard to see. Try to reduce the precision of these values down to one or two decimals and get them visible on screen. Just don't worry about making it perfect.

File: frontend/src/components/Telemetry.js (hunk 3 above)
Comment (john-langlois) [reply frontend/src/components/Telemetry.js]: Yes I noticed that, updated it and going to just pass it through telemetryStatus

**Review by jeffstjean — APPROVED** [review]

**Pull request merged**